src/savedata.c

```diff
@@ -845,585 +838,558 @@ int SaveTableEntry_BodySize (int saveTableID)
+        pageInfo[i].blockID = saveTable[i].blockID;
 
-        v2 += param0[v1].size;
+        totalSize += pageInfo[i].size;
 
-        if ((v1 == gSaveTableSize - 1) || (v0[v1].blockID != v0[v1 + 1].blockID)) {
-            v2 += sizeof(SaveBlockFooter);
+        if ((i == gSaveTableSize - 1) || (saveTable[i].blockID != saveTable[i + 1].blockID)) {
+            totalSize += sizeof(SaveBlockFooter);
         }
     }
 
-    if (v1 % 4 != 0) {
-    }
-
-    GF_ASSERT(v2 <= 0x1000 * 32);
+    GF_ASSERT(totalSize <= SAVE_SECTOR_SIZE * SAVE_PAGE_MAX);
 }
 
-static void sub_02025258 (SaveBlockInfo * param0, const SavePageInfo * param1)
+static void SaveBlockInfo_Init (SaveBlockInfo *blockInfo, const SavePageInfo *pageInfo)
 {
-    int v0 = 0;
-    int v1, v2;
-    int v3, v4;
+    int totalSectors = 0, blockOffset = 0, block, page = 0;
 
-    v1 = 0;
-    v2 = 0;
-    v4 = 0;
+    for (block = 0; block < SAVE_BLOCK_ID_MAX; block++) {
+        blockInfo[block].saveBlockID = block;
+        blockInfo[block].size = 0;
 
-    for (v3 = 0; v3 < 2; v3++) {
-        param0[v3].saveBlockID = v3;
-        param0[v3].size = 0;
-
-        for (; param1[v4].blockID == v3 && v4 < gSaveTableSize; v4++) {
-            param0[v3].size += param1[v4].size;
+        for (; pageInfo[page].blockID == block && page < gSaveTableSize; page++) {
+            blockInfo[block].size += pageInfo[page].size;
         }
 
-        param0[v3].size += sizeof(SaveBlockFooter);
-        param0[v3].sectorStartPos = v1;
-        param0[v3].offset = v2;
-        param0[v3].sectorsInUse = (param0[v3].size + 0x1000 - 1) / 0x1000;
+        blockInfo[block].size += sizeof(SaveBlockFooter);
+        blockInfo[block].sectorStartPos = totalSectors;
+        blockInfo[block].offset = blockOffset;
+        blockInfo[block].sectorsInUse = (blockInfo[block].size + SAVE_SECTOR_SIZE - 1) / SAVE_SECTOR_SIZE;
 
-        v1 += param0[v3].sectorsInUse;
-        v2 += param0[v3].size;
+        totalSectors += blockInfo[block].sectorsInUse;
+        blockOffset += blockInfo[block].size;
     }
 
-    GF_ASSERT(v1 == param0[2 - 1].sectorStartPos + param0[2 - 1].sectorsInUse);
-    GF_ASSERT(v1 <= 32);
+    GF_ASSERT(totalSectors == blockInfo[SAVE_BLOCK_ID_MAX - 1].sectorStartPos + blockInfo[SAVE_BLOCK_ID_MAX - 1].sectorsInUse);
+    GF_ASSERT(totalSectors <= SAVE_PAGE_MAX);
 }
 
-static void sub_020252EC (SaveDataBody * param0, const SavePageInfo * param1)
+static void SaveTable_Clear (SaveDataBody *body, const SavePageInfo *pageInfo)
 {
-    const SaveTableEntry * v0 = gSaveTable;
-    int v1;
-    int v2;
-    void * v3;
-    u32 v4;
-
-    MI_CpuClearFast(param0->data, sizeof(param0->data));
-
-    for (v1 = 0; v1 < gSaveTableSize; v1++) {
-        v4 = param1[v1].location;
-        v3 = &param0->data[v4];
-        v2 = param1[v1].size;
-        MI_CpuClearFast(v3, v2);
-        v0[v1].initFunc(v3);
+    const SaveTableEntry *saveTable = gSaveTable;
+    int i, size;
+    void *page;
+    u32 location;
```

Comment:
Do these need to declared C89-style, or can they be declared inline?

Applies to other instances in the file.